File: Nature.cpp
#include <string>
#include <map>
#include "Nature.h"

Nature::Nature(std::string natureInput, std::string multiplierInput, std::string divisorInput) {
    _name = natureInput;
    if (multiplierInput != divisorInput) {
        _stats[multiplierInput] = 1.1;
        _stats[divisorInput] = 0.9;
    }
};

std::map<std::string, Nature> natureMap = {
    {"Hardy", Nature("Hardy", "attack", "attack")},
    {"Lonely", Nature("Lonely", "attack", "defense")},
    {"Brave", Nature("Brave", "attack", "speed")},
    {"Adamant", Nature("Adamant", "attack", "specialAttack")},
    {"Naughty", Nature("Naughty", "attack", "specialDefense")},
    {"Bold", Nature("Bold", "defense", "attack")},
    {"Docile", Nature("Docile", "defense", "defense")},
    {"Relaxed", Nature("Relaxed", "defense", "speed")},
    {"Impish", Nature("Impish", "defense", "specialAttack")},
    {"Lax", Nature("Lax", "defense", "specialDefense")},
    {"Timid", Nature("Timid", "speed", "attack")},
    {"Hasty", Nature("Hasty", "speed", "defense")},
    {"Serious", Nature("Serious", "speed", "speed")},
    {"Jolly", Nature("Jolly", "speed", "specialAttack")},
    {"Naive", Nature("Naive", "speed", "specialDefense")},
    {"Modest", Nature("Modest", "specialAttack", "attack")},
    {"Mild", Nature("Mild", "specialAttack", "defense")},
    {"Quiet", Nature("Quiet", "specialAttack", "speed")},
    {"Bashful", Nature("Bashful", "specialAttack", "specialAttack")},
    {"Rash", Nature("Rash", "specialAttack", "specialDefense")},
    {"Calm", Nature("Calm", "specialDefense", "attack")},
    {"Gentle", Nature("Gentle", "specialDefense", "defense")},
    {"Sassy", Nature("Sassy", "specialDefense", "speed")},
    {"Careful", Nature("Careful", "specialDefense", "specialAttack")},
    {"Quirky", Nature("Quirky", "specialDefense", "specialDefense")}
};

File: Pokemon Black2 Replica/Assets/Pokemon/Nature.cs
using System.Collections.Generic;
using UnityEngine;

public class Nature
{
    // Automatic Properties
    public string Name { get; }
    public Dictionary<string, double> StatMultipliers { get; } = new Dictionary<string, double>
    {
        {"Attack", 1.0},
        {"Defense", 1.0},
        {"Speed", 1.0},
        {"SpecialAttack", 1.0},
        {"SpecialDefense", 1.0}
    };

    // Constructor
    Nature(string inputNatureName, string inputMultiplier, string inputDivisor)
    {
        Name = inputNatureName;
        if (inputMultiplier != inputDivisor)
        {
            StatMultipliers[inputMultiplier] = 1.1;
            StatMultipliers[inputDivisor] = 0.9;
        }
    }

    // Establish Natures
    public static Dictionary<string, Nature> NatureMap { get; }

    static Nature()
    {
        NatureMap = new Dictionary<string, Nature>
        {
            {"Hardy", new Nature("Hardy", "Attack", "Attack")},
            {"Lonely", new Nature("Lonely", "Attack", "Defense")},
            {"Brave", new Nature("Brave", "Attack", "Speed")},
            {"Adamant", new Nature("Adamant", "Attack", "SpecialAttack")},
            {"Naughty", new Nature("Naughty", "Attack", "SpecialDefense")},
            {"Bold", new Nature("Bold", "Defense", "Attack")},
            {"Docile", new Nature("Docile", "Defense", "Defense")},
            {"Relaxed", new Nature("Relaxed", "Defense", "Speed")},
            {"Impish", new Nature("Impish", "Defense", "SpecialAttack")},
            {"Lax", new Nature("Lax", "Defense", "SpecialDefense")},
            {"Timid", new Nature("Timid", "Speed", "Attack")},
            {"Hasty", new Nature("Hasty", "Speed", "Defense")},
            {"Serious", new Nature("Serious", "Speed", "Speed")},
            {"Jolly", new Nature("Jolly", "Speed", "SpecialAttack")},
            {"Naive", new Nature("Naive", "Speed", "SpecialDefense")},
            {"Modest", new Nature("Modest", "SpecialAttack", "Attack")},
            {"Mild", new Nature("Mild", "SpecialAttack", "Defense")},
            {"Quiet", new Nature("Quiet", "SpecialAttack", "Speed")},
            {"Bashful", new Nature("Bashful", "SpecialAttack", "SpecialAttack")},
            {"Rash", new Nature("Rash", "SpecialAttack", "SpecialDefense")},
            {"Calm", new Nature("Calm", "SpecialDefense", "Attack")},
            {"Gentle", new Nature("Gentle", "SpecialDefense", "Defense")},
            {"Sassy", new Nature("Sassy", "SpecialDefense", "Speed")},
            {"Careful", new Nature("Careful", "SpecialDefense", "SpecialAttack")},
            {"Quirky", new Nature("Quirky", "SpecialDefense", "SpecialDefense")}
        };
    }
}
